src/low_points.ts

```diff
@@ -0,0 +1,44 @@
+export function isLowPoint(prev: number, cur: number, next: number): boolean {
+  return (prev > cur) && (next > cur);
+}
+
+// Assumes that the grid has a consistent width and height.
+export function locateLowPointsInGrid(grid: number[][]): [number, number][] {
+  let maximum = 9; // TODO: autogenerate this
+  let paddingHorizontal: number[] = Array(grid[0].length).fill(maximum);
+
+  // Builds a grid with padding around it
+  grid = [paddingHorizontal].concat(grid).concat([paddingHorizontal]);
+  grid = grid.map(line => [maximum].concat(line).concat([maximum]));
+
+  let locationsInPaddedGrid = [];
```

Comment:
Instead of using a doubly nest array, consider creating an interface for the coordinates like so
```typescript
interface Point {
  x: number
  y: number
}
```